src/Forms/NumericField.php

```diff
@@ -149,28 +178,36 @@ public function Value()
         return $formatter->format($this->value, $this->getNumberType());
     }
 
-    public function setValue($value, $data = null)
+    public function getValueForValidation(): mixed
     {
-        $this->originalValue = $value;
-        $this->value = $this->cast($value);
-        return $this;
+        $value = $this->value;
+        if (is_numeric($value) && is_string($value)) {
+            return $this->cast($value);
+        }
+        return $value;
     }
 
     /**
      * Helper to cast non-localised strings to their native type
-     *
-     * @param string $value
-     * @return float|int
      */
-    protected function cast($value)
+    protected function cast(mixed $value): mixed
     {
-        if (strlen($value ?? '') === 0) {
+        if ($value === false) {
+            return false;
+        }
```

Comment:
What situation is this happening in? Why `false` explicitly and not `is_boolean()`?

Reply:
`false` is set when when the localisation formatter fails to parse the value in setSubmittedValue()